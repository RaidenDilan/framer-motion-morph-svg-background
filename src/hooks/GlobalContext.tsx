import React, { createContext, ReactNode, useContext, useState } from 'react';

interface GlobalContextInterface {
  isLoading: boolean;
}

export const defaultValues: GlobalContextInterface = {
  isLoading: false,
};

export const GlobalCtx = createContext<GlobalContextInterface>(defaultValues);

export const GlobalProvider = ({ children }: { children: ReactNode }) => {
  const [isLoading, setIsLoading] = useState<boolean>(false);

  return (
    <GlobalCtx.Provider value={{ isLoading }}>{children}</GlobalCtx.Provider>
  );
};

export const useGlobalContext = () => useContext(GlobalCtx);
