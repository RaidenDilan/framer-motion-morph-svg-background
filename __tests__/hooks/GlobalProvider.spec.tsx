import React from 'react';
import { render } from '@testing-library/react';
import { GlobalProvider } from '../../src/hooks/GlobalContext';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => {
  const requireActual = jest.requireActual('react-router-dom');
  return {
    ...requireActual,
    useNavigate: () => mockedUsedNavigate,
  };
});

describe(__filename, () => {
  it('renders the global provider', () => {
    const { asFragment } = render(
      <GlobalProvider children={<React.Fragment />} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
