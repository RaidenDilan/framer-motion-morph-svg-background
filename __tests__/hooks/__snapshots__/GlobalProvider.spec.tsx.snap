// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Users/Raiden/development/.boilerplates/vite-boilerplate/__tests__/hooks/GlobalProvider.spec.tsx renders the global provider 1`] = `<DocumentFragment />`;
